pkg/registry/authorization/selfsubjectaccessreview/rest.go

```diff
@@ -72,6 +76,28 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 		}
 	}
 
+	var userExtraCopy map[string][]string
+	if userFromCtx.GetExtra() != nil {
+		userExtraCopy = make(map[string][]string)
+		for k, v := range userFromCtx.GetExtra() {
+			userExtraCopy[k] = v
+		}
+	}
+
+	for _, scope := range userFromCtx.GetExtra()[authorizationv1.ScopesKey] {
+		if scope == authorizationscope.UserAccessCheck {
+			userExtraCopy[authorizationv1.ScopesKey] = []string{authorizationscope.UserFull}
```

Comment:
why should all the scopes not be honored?

Reply:
If you have multiple scopes, one allowing you to check access and another restricting you to a single namespace, this change will make the self-SAR come back true for another namespace, correct?

Reply:
You're right, good point -- I'll fix it so that we also take into account any `role:` scopes, too.

Reply:
@deads2k I've updated the code to:
- also consider role scopes before adding `user:full` to a self-SAR request
- when needed, extend the list of scopes with `user:full` instead of replacing the whole list